draft-ietf-oauth-device-flow.xml

```diff
@@ -415,33 +468,169 @@
 
-  <t>The error code "authorization_pending" and "slow_down" are considered
-    soft errors. The the client should continue to poll when
-    receiving "authorization_pending"
-    errors, reducing the interval if a "slow_down" error is received.
-    Other error codes are considered hard errors, the client should stop polling
-    and react accordingly, for example, by displaying an error to the user.
+  <t>The error codes <spanx style="verb">authorization_pending</spanx> and
+    <spanx style="verb">slow_down</spanx> are considered soft errors. The client
+    should continue to poll the token endpoint by repeating the Device
+    Token Request (<xref target="TokenRequest" />) when receiving soft errors,
+    increasing the poll interval if a <spanx style="verb">slow_down</spanx> error
+    is received. Other error codes are considered hard errors, the client should
+    stop polling and react accordingly, for example, by displaying an error to
+    the user.
+  </t>
+
+  <t>
+    The interval at which the client polls MUST NOT be more frequent than the
+    <spanx style="verb">interval</spanx> parameter returned in the Device Authorization
+    Response (see <xref target="AuthorizationResponse"/>).
+  </t>
+
+  <t>
+  The assumption of this specification is that the secondary device the
+  user is authorizing the request on does not have a way to communicate back
+  to the OAuth client. That only a one-way channel is required makes this flow
+  useful in many scenarios such as a HTML application on TV that can only 
+  make outbound requests. If a return channel were to exist for the chosen user
+  interaction interface, then the device MAY wait until notified on that channel
+  that the user has completed the action before initiating the token request.
+  Such behavior however is outside the scope of this specification.
   </t>
 
   </section>
 
 
     </section> 
 
+    <section anchor="Discovery" title="Discovery Metadata">
+      <t>
+        Support for the device flow MAY be declared in the OAuth 2.0
+        Authorization Server Metadata <xref target="I-D.ietf-oauth-discovery"/>
+        with the following metadata:
+      </t>
+      <t>
+      <list style="hanging"> 
+        <t hangText="device_authorization_endpoint">
+          <vspace/>
+          OPTIONAL.
+          URL of the authorization server's device authorization endpoint 
+          defined in <xref target="AuthorizationRequest"/>.
+        </t>
+      </list>
+      </t>
+    </section>
+
+    <section anchor="Security" title="Security Considerations">
+        <section title="User Code Brute Forcing" anchor="UserCodeBruteForce">
+          <t>
+          Since the user code is typed by the user, the entropy is typically
+          less than would be used for the device code or other OAuth bearer token types.
+          It is therefore recommended that the server rate-limit user code
+          attempts. The user code SHOULD have enough entropy that when combined
```

Comment:
Is the "should" appropriate here? --> The user code SHOULD have enough entropy

Wouldn't it be more appropriate to use the term MUST?